dist/templates/rbac-ovnkube-master.yaml.j2

```diff
@@ -191,3 +191,42 @@ spec:
         apiVersions: ["v1"]
         operations:  ["UPDATE"]
         resources:   ["namespaces"]
+
+{% if ovn_pre_conf_udn_addr_enable == "true" -%}
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: default-network-annotation
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        operations:  ["UPDATE"]
+        resources:   ["pods"]
+  failurePolicy: Fail
+  validations:
+    # Prevent any changes to the default-network annotation after pod creation:
+    # - If annotation exists in old pod: new pod must have same annotation with identical value
+    # - If annotation doesn't exist in old pod: new pod must also not have it
+    - expression: >
+        ('v1.multus-cni.io/default-network' in oldObject.metadata.annotations)
+        ? ('v1.multus-cni.io/default-network' in object.metadata.annotations) && oldObject.metadata.annotations['v1.multus-cni.io/default-network'] == object.metadata.annotations['v1.multus-cni.io/default-network']
+        : !('v1.multus-cni.io/default-network' in object.metadata.annotations)
```

Comment:
nit: 
The trinary operator confuses me in this case, the following expression should also work. Both are good  your call :slightly_smiling_face: 
```
!('v1.multus-cni.io/default-network' in object.metadata.annotations) ||
        ('v1.multus-cni.io/default-network' in oldObject.metadata.annotations) &&
          oldObject.metadata.annotations['v1.multus-cni.io/default-network'] == object.metadata.annotations['v1.multus-cni.io/default-network']
```

Reply:
Doesn't the suggested CEL allow for annotation removal? we don't want that.

Reply:
@kyrtapz the code looks fine to me, but I am confused with your comment `This rule doesn't block modifying the value of the annotation, it just blocks adding or removing it.` ->
but your rule actually does block value change and I thought that's what we want?